src/ServicePulse.Host/vue/test/specs/monitoring/filtering-endpoints.spec.ts

```diff
@@ -1,32 +1,319 @@
+          "Universe.Solarsystem.Venus.Endpoint3",
+          "Universe.Solarsystem.Venus.Endpoint4",
+          "Universe.Solarsystem.Earth.Endpoint5",
+          "Universe.Solarsystem.Earth.Endpoint6",
+        ])
+      );
+
+      //Act
+      await driver.goTo("monitoring");
+      await groupEndpointsBy({ numberOfSegments: 3 });
+      expect(endpointGroupNames()).toEqual(["Universe.Solarsystem.Earth", "Universe.Solarsystem.Mercury", "Universe.Solarsystem.Venus"]);
+      expect(endpointGroup("Universe.Solarsystem.Mercury").Endpoints).toEqual(["Endpoint1", "Endpoint2"]);
+      expect(endpointGroup("Universe.Solarsystem.Venus").Endpoints).toEqual(["Endpoint3", "Endpoint4"]);
+      expect(endpointGroup("Universe.Solarsystem.Earth").Endpoints).toEqual(["Endpoint5", "Endpoint6"]);
+      await enterFilterString("WrongName");
+
+      //Assert
+      await waitFor(() => expect(endpointGroupNames()).toEqual([]));
+      await waitFor(() => expect(endpointWithName("Endpoint1")).toBeNull());
+      await waitFor(() => expect(endpointWithName("Endpoint2")).toBeNull());
+      await waitFor(() => expect(endpointWithName("Endpoint3")).toBeNull());
+      await waitFor(() => expect(endpointWithName("Endpoint4")).toBeNull());
+      await waitFor(() => expect(endpointWithName("Endpoint5")).toBeNull());
+      await waitFor(() => expect(endpointWithName("Endpoint6")).toBeNull());
+    });
+  });
+
+  describe("Rule: Filter string can get and set the filter parameter in the permalink", () => {
+    it("Example: The permalink has the filter parameter set to match only 1 endpoint", async ({ driver }) => {
+      //Arrange
+      await driver.setUp(precondition.serviceControlWithMonitoring);
+      await driver.setUp(precondition.monitoredEndpointsNamed(["Universe.Solarsystem.Earth.Endpoint1", "Universe.Solarsystem.Earth.Endpoint2", "Universe.Solarsystem.Earth.Endpoint3"]));
+
+      //Act
+      await driver.goTo("monitoring?filter=Endpoint1");
+
+      //Assert
+      await waitFor(() => expect(filteredByName("Endpoint1")).toBeInTheDocument());
+      await waitFor(() => expect(endpointWithName("Universe.Solarsystem.Earth.Endpoint1")).toBeInTheDocument());
+      await waitFor(() => expect(endpointWithName("Universe.Solarsystem.Earth.Endpoint2")).toBeNull());
+      await waitFor(() => expect(endpointWithName("Universe.Solarsystem.Earth.Endpoint3")).toBeNull());
+    });
+
+    it("Example: The permalink does not have the filter parameter set and then a filter string is entered that matches only 1 endpoint", async ({ driver }) => {
+      //Arrange
+      await driver.setUp(precondition.serviceControlWithMonitoring);
+      await driver.setUp(precondition.monitoredEndpointsNamed(["Universe.Solarsystem.Earth.Endpoint1", "Universe.Solarsystem.Earth.Endpoint2", "Universe.Solarsystem.Earth.Endpoint3"]));
+
+      //Act
+      await driver.goTo("monitoring?filter=Endpoint1");
+      let foo = await driver.getRouter();
```

Comment:
Although exposing the Vue router is very convenient, it ties the test to Vue, which we want to avoid. 

- We could code assertions against the window.location instead. 
- Another option is to continue using vue router but abstracted away, hidden in the driver, but never return the whole driver itself, make the driver have a method with a name like "CurrentPath" or something like that, that only returns a string.

Reply:
Using waitFor with window.location seems to work.